### PR TITLE
Infra 3710 edge cases WIP

### PR DESCRIPTION
**Pull request by elainearbaugh**
unusual cases
more tests
clean up

More specific explanations in my comments. Actual changes are in https://github.com/Affirm/cabot/pull/50/commits/1287498bfc56e363a05694f5f9911461a517547a

### CODE REVIEW

**Comment by elainearbaugh** [comment]
trigger build

File: cabot/templates/cabotapp/statuscheckresult_detail.html [hunk 1]
```diff
@@ -55,6 +55,8 @@
   label = s.series
   labels.push label
   for point in s.datapoints
+    if point[1] == 'NaN'
+      point[1] = 0
```
Comment (elainearbaugh) [inline cabot/templates/cabotapp/statuscheckresult_detail.html]: NaNs break the graphs

File: cabot/metricsapp/api/grafana_elastic.py [hunk 2]
```diff
@@ -192,3 +197,29 @@ def get_es_status_check_fields(dashboard_info, panel_info, series_list):
         fields['queries'].append(query)
 
     return fields
+
+
+def adjust_time_range(queries, time_range):
```
Comment (elainearbaugh) [inline cabot/metricsapp/api/grafana_elastic.py]: Change the query if the user specifies a smaller time range so we don't query for data we don't need

File: cabot/metricsapp/api/grafana_elastic.py [hunk 3]
```diff
@@ -21,11 +22,15 @@ def _get_terms_settings(agg):
     settings = agg['settings']
     order_by = settings.get('orderBy')
     if order_by:
+        # Grafana indicates sub-aggregation ordering by a number representing the aggregation
```
Comment (elainearbaugh) [inline cabot/metricsapp/api/grafana_elastic.py]: Unsupported aggregation case, handle errors more gracefully

File: cabot/metricsapp/api/grafana.py [hunk 4]
```diff
@@ -3,27 +3,38 @@
...
 from django.core.exceptions import ValidationError
 from pytimeparse import parse
+from urlparse import urljoin
 
 
 logger = logging.getLogger(__name__)
 
 
-def get_dashboards(grafana_instance):
+def _grafana_api_request(grafana_instance, uri):
```
Comment (elainearbaugh) [inline cabot/metricsapp/api/grafana.py]: Move common functionality for requests to Grafana API into a helper function

File: cabot/templates/cabotapp/statuscheckresult_detail.html (hunk 1 above)
Comment (mahendra) [inline cabot/templates/cabotapp/statuscheckresult_detail.html]: :-)
